pennylane/gradients/gradient_transform.py

```diff
@@ -327,14 +327,18 @@ def jacobian_wrapper(
 
             # Special case where we apply a Jax transform (jacobian e.g.) on the gradient transform and argnums are
             # defined on the outer transform and therefore on the args.
-            if interface == "jax":
-                argnum_cjac = trainable_params or argnums
+            if argnums is None and argnums_ is None and interface == "jax":
+                cjac = qml.transforms.classical_jacobian(
+                    qnode, argnum=qml.math.get_trainable_indices(args), expand_fn=self.expand_fn
+                )(*args, **kwargs)

```

Comment:
Couldn't we absorb this branch into the logic below as well, just defining 
```suggestion
            if argnums is None and argnums_ is None and interface == "jax":
                argnum_cjac=qml.math.get_trainable_indices(args)
```
here and using the call to `classical_jacobian` below?

Reply:
I think that we have merge conflict here we should not need to change anything in the gradient transform

Reply:
Oh, I'm sorry about that. I thought I had figured out the change.